### PR TITLE
add support for townlong-yak addons

### LINKED ISSUE #512: TownLong-Yak Addons Repository
The TownLong-Yak addons (OPie, etc.) aren't kept up to date in any of the standard repositories (OPie has been several versions behind for months now, and is missing bug fixes and a lot of Shadowlands functionality).

These addons are quite popular, TownLong-Yak is a pretty established WoW resource, and they ought to be added to Ajour as a repository.

The site has an API to check for updates: https://www.townlong-yak.com/addons/about/update-api

### PR DESCRIPTION
**Pull request by tarkah**
Resolves #512 

## Proposed Changes
  - Add townlong-yak addon support via the hub api

## Checklist

- [x] Tested on Windows
- [x] Tested on MacOS
- [ ] Tested on Linux
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

File: crates/core/src/cache.rs [hunk 1]
```diff
@@ -207,13 +208,23 @@ impl TryFrom<&Addon> for AddonCacheEntry {
...
 
+            let external_release_id = if repository == RepositoryKind::TownlongYak {
+                addon.file_id().map(ExternalReleaseId::FileId)
+            } else {
+                addon
+                    .version()
+                    .map(str::to_string)
+                    .map(ExternalReleaseId::Version)
+            };
+
```
Comment (tarkah) [inline crates/core/src/cache.rs:219]: This is how we cache the version that was just installed / updated. When we reload the addon from the cache, it'll use this as the version / file_id to check against the remote API. This resolves issues where the addon .toc version is not comparable to the remote version on the API

File: crates/core/src/catalog.rs [hunk 2]
```diff
@@ -7,7 +7,7 @@ use chrono::prelude::*;
 use isahc::ResponseExt;
 use serde::{Deserialize, Serialize};
 
-const CATALOG_URL: &str = "https://github.com/ajour/ajour-catalog/raw/master/catalog.json";
+const CATALOG_URL: &str = "https://raw.githubusercontent.com/ajour/ajour-catalog/3ce1414498adbc7167f11fe0dac7cd58e7b55391/catalog.json";
```
Comment (tarkah) [inline crates/core/src/catalog.rs:10]: Make sure we change this back before merging. I switched over to test.

File: crates/core/src/parse.rs [hunk 3]
```diff
@@ -688,7 +731,20 @@ fn build_addons(
...
+            // validate the installed version against the remote version
+            match &e.external_release_id {
+                Some(ExternalReleaseId::FileId(file_id)) => {
+                    repo_package.metadata.file_id = Some(*file_id)
+                }
+                Some(ExternalReleaseId::Version(version)) => {
+                    repo_package.metadata.version = Some(version.clone())
+                }
+                None => {}
+            }
```
Comment (tarkah) [inline crates/core/src/parse.rs:747]: Here is where we apply that cached remote release version when building addon from cache entry

File: src/gui/update.rs [hunk 4]
```diff
@@ -757,15 +757,16 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
...
-                    version = Some(package.version);
-                }
-                if let Some(version) = version {
-                    addon.set_version(version);
+                    addon.set_version(package.version);
+
+                    if let Some(file_id) = package.file_id {
+                        addon.set_file_id(file_id);
+                    }
                 }
```
Comment (tarkah) [inline src/gui/update.rs:767]: Here is where we set version / file id from the newly installed / updated package. This is done prior to updating the cache entry, so cache will always stay up to date with latest external release id when installing / updating

File: src/gui/mod.rs [hunk 5]
```diff
@@ -1838,6 +1844,7 @@ impl std::fmt::Display for CatalogSource {
                 catalog::Source::Curse => "Curse",
                 catalog::Source::Tukui => "Tukui",
                 catalog::Source::WowI => "WowInterface",
+                catalog::Source::TownlongYak => "TownlongYak",
```
Comment (casperstorm) [inline src/gui/mod.rs:1847]: Wondering if we should write: Townlong Yak

**Comment by casperstorm** [comment]
I've tested this on both Windows and macOS and it works perfectly.
I don't really have any comments  to the code. Its clean. Good job, once again mate.

File: src/gui/mod.rs (hunk 5 above)
Comment (tarkah) [reply src/gui/mod.rs:1847]: Might as well put a `-` in at that point?

Comment (casperstorm) [reply src/gui/mod.rs:1847]: Yeah lets do that.

**Pull request merged**